x-pack/plugins/security_solution/public/cases/components/case_action_bar/status_context_menu.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useCallback, useMemo, useState } from 'react';
+import { memoize } from 'lodash/fp';
+import { EuiPopover, EuiContextMenuPanel, EuiContextMenuItem } from '@elastic/eui';
+import { CaseStatus } from '../../../../../case/common/api';
+import { Status, statuses } from '../status';
+
+interface Props {
+  currentStatus: CaseStatus;
+  onStatusChanged: (status: CaseStatus) => void;
+}
+
+const StatusContextMenuComponent: React.FC<Props> = ({ currentStatus, onStatusChanged }) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const closePopover = useCallback(() => setIsPopoverOpen(false), []);
+  const openPopover = useCallback(() => setIsPopoverOpen(true), []);
+  const popOverButton = useMemo(
+    () => <Status type={currentStatus} withArrow onClick={openPopover} />,
+    [currentStatus, openPopover]
+  );
+
+  const onContextMenuItemClick = useMemo(
+    () =>
+      memoize<(status: CaseStatus) => () => void>((status) => () => {
+        closePopover();
+        onStatusChanged(status);
+      }),
+    [closePopover, onStatusChanged]
+  );
+
+  const caseStatuses = Object.keys(statuses) as CaseStatus[];
+  const panelItems = caseStatuses.map((status: CaseStatus) => (
+    <EuiContextMenuItem
+      key={status}
+      icon={status === currentStatus ? 'check' : 'empty'}
+      onClick={onContextMenuItemClick(status)}
```

Comment:
```suggestion
      onClick={() => onContextMenuItemClick(status)}
```
need to make this an anon/lambda function or `onContextMenuItemClick` will get invoked on render for each menu item

Reply:
`onContextMenuItemClick` function returns another function that is `memoized`. This is a trick to avoid the callback and potential rerenders.

Reply:
awesome!! bookmarking this for the future ;)